### PR TITLE
Fixes #992: Updates account implementation for a-c 47.0

### LINKED ISSUE #992: Fix failing master
Looks like [a change with a-c 47](https://github.com/mozilla-mobile/android-components/commit/54da16624452b2db3a6d6276b1549582897c3429) broke our implementation of sync. We should get master up and running again.

### PR DESCRIPTION
**Pull request by sblatz**
Note: the implementation for onEditListener will be added in #929.

I tested logging in and it seems to work properly.

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarUIView.kt [hunk 1]
```diff
@@ -54,6 +54,9 @@ class ToolbarUIView(
             hintColor = ContextCompat.getColor(context, R.color.search_text)
 
             setOnEditListener(object : mozilla.components.concept.toolbar.Toolbar.OnEditListener {
+                override fun onCancelEditing(): Boolean {
+                    TODO("not implemented")
```
Comment (boek) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarUIView.kt]: Will this cause any problems?

Comment (pocmo) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarUIView.kt]: Yes, TODO() throws an exception :)

**Comment by pocmo** [comment]
FYI @grigoryk :)

File: app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarUIView.kt (hunk 1 above)
Comment (sblatz) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarUIView.kt]: I would like to fast-follow this with #929 landing.

**Pull request merged**